trunner/harness/psh.py

```diff
@@ -81,4 +94,11 @@ def __call__(self, result: TestResult) -> TestResult:
                 ) from e
 
         result.set_stage(TestStage.RUN)
-        return self.next_harness(result)
+        test_result = self.next_harness(result)
+
+        # Re-enabling log output to release collected output
+        if self.suppress_dmesg:
+            self.dut.pexpect_proc.sendline("dmesg -E")
+            self.dut.expect_exact(self.prompt)
```

Comment:
missing `prompt_timeout support` and we should translate `pexpect` exceptions to `ShellError` (see above)